src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -518,7 +523,15 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
   // same TaskID to request a worker
   auto resource_spec_msg = scheduling_key_entry.resource_spec.GetMutableMessage();
   resource_spec_msg.set_task_id(TaskID::FromRandom(job_id_).Binary());
-  const TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
+  TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
+
+  // Copy runtime env rescheduling context.
+  if (reply_last_time != nullptr) {
+    resource_spec.GetMutableMessage()
```

Comment:
This is awkward to be updated in this function. Can we update it before calling this method? This must be updated within the callback of `lease_client->RequestWorkerLease`